### ISSUE TITLE
Apply version catalog

### ISSUE THREAD

**Issue by ndr-brt** [eclipse-edc/Samples#39]
# Feature Request

We [decided to use gradle version catalogs](https://github.com/eclipse-edc/Connector/tree/main/docs/developer/decision-records/2022-10-21-gradle-versioncatalogs), so we need to use them also in this repository.

## Why Is the Feature Desired?
_Are there any requirements?_

## Solution Proposal
_If possible, provide a (brief!) solution proposal._

## Type of Issue
_i.e., new feature, improvement, cleanup, etc._

**Comment by robertdevlin-ms** [comment]
Similarly, while working on another sample, we discovered that there is a gradle bug for versions <7.6. 
https://github.com/FasterXML/jackson-core/issues/955

It would result in this error locally:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'samples'.
> java.util.concurrent.ExecutionException: org.gradle.api.GradleException: Failed to create Jar file /home/censored/.gradle/caches/jars-9/94ac7f0b504e3b47bc2eec90c/jackson-core-2.15.0.jar.
```

Currently as a workaround locally, updating the distributionUrl in Samples/gradle/wrapper/gradle-wrapper.properties to 7.6.1 solves this issue. I think it makes sense for this pull request to also update all the way to version 8 to match the Connector. https://github.com/eclipse-edc/GradlePlugins/pull/89

Syncing the version catalog changes along with the gradle version would bring the Samples repository closer to the Connector, and from a dependency-management perspective I believe the version catalog and updating gradle would make sense to do together. Leaving a comment just in case anyone else comes across this before a PR comes out.

**Comment by ndr-brt** [comment]
There's an issue related: #45 the two things could be tackled in different PRs

**Comment by ndr-brt** [comment]
closed by #47
